src/main/java/com/duke/bot/Duke.java

```diff
@@ -0,0 +1,157 @@
+            case "list":
+                ui.print(tasks.printList());
+                ui.printEmptyLine();
+                ui.setToken("\\p{javaWhitespace}+");
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            case "done":
+                int targetIdx = ui.getNextInt() - 1;
+                this.tasks.getTask(targetIdx).markDone();
+                ui.print("Good job! I've marked this task as done:");
+                ui.print(String.format("%d. %s\n", targetIdx + 1, tasks.getTask(targetIdx).toString()));
+                ui.print("\n");
+                ui.setToken("\\p{javaWhitespace}+");
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            case "todo":
+                ui.setToken("\\n");
+                String userInput = ui.getNextLine();
+                if (userInput.equals("")) {
+                    ui.resetScanner();
+                    throw new DukeException("Uh-oh! Description of todo cannot be empty!");
+                }
+                TodoTask todoTask = TodoTask.createTodoTask(userInput);
+                tasks.addTask(todoTask);
+                ui.print("Added: " + todoTask.toString());
+                ui.print(String.format("Now you have %d task(s) on your list.\n", tasks.getSize()));
+                ui.printEmptyLine();
+                //echo(new Scanner(System.in);
+                ui.setToken("\\p{javaWhitespace}+");
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            case "deadline":
+                ui.setToken("\\s*/by\\s*|\\n");
+                String action = ui.getNext();
+                String dateInput = ui.getNext();
+                LocalDate byDate = LocalDate.parse(dateInput);
+                DeadlineTask deadlineTask = DeadlineTask.createDeadlineTask(action, byDate);
+                tasks.addTask(deadlineTask);
+                ui.print("Added: " + deadlineTask.toString());
+                ui.print(String.format("Now you have %d task(s) on your list.\n", tasks.getSize()));
+                ui.printEmptyLine();
+                ui.setToken("\\p{javaWhitespace}+");
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            case "event":
+                ui.setToken("\\s*/at\\s*|\\n");
+                action = ui.getNext();
+                dateInput = ui.getNext();
+                LocalDate atDate = LocalDate.parse(dateInput);
+                EventTask eventTask = EventTask.createEventTask(action, atDate);
+                tasks.addTask(eventTask);
+                ui.print("Added: " + eventTask.toString());
+                ui.print(String.format("Now you have %d task(s) on your list.\n", tasks.getSize()));
+                ui.printEmptyLine();
+                ui.setToken("\\p{javaWhitespace}+");
```

Comment:
I would suggest grouping these few lines into a method in DukeUi class which can make the code more readable.